src/main/kotlin/com/example/ajaxproject/pbp/NotificationAnnotationBeanPostProcessor.kt

```diff
@@ -0,0 +1,99 @@
+package com.example.ajaxproject.pbp
+
+import com.example.ajaxproject.config.Notification
+import com.example.ajaxproject.dto.request.EmailDTO
+import com.example.ajaxproject.dto.request.GroupChatDTO
+import com.example.ajaxproject.repository.GroupChatRoomRepository
+import com.example.ajaxproject.service.interfaces.EmailSenderService
+import org.bson.types.ObjectId
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.cglib.proxy.InvocationHandler
+import org.springframework.cglib.proxy.Proxy
+import org.springframework.stereotype.Component
+import org.springframework.stereotype.Service
+import java.lang.reflect.Method
+import kotlin.reflect.KClass
+import kotlin.reflect.full.memberFunctions
+
+@Component
+class NotificationAnnotationBeanPostProcessor(
+    @Qualifier("javaMailEmailSenderService") private val emailSenderService: EmailSenderService
+) : BeanPostProcessor {
+
+    @Autowired
+    private lateinit var groupChatRoomRepository: GroupChatRoomRepository
```

Comment:
why mixed constructor injection and field injection?

Reply:
Really did it really illogically) Rewrote fully in constructor